examples/wifi-echo/server/esp32/main/include/RendezvousDeviceDelegate.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef __RENDEZVOUSDEVICEDELEGATE_H__
+#define __RENDEZVOUSDEVICEDELEGATE_H__
+
+#include <platform/CHIPDeviceLayer.h>
+#include <transport/RendezvousSession.h>
+
+using namespace ::chip;
+
+class RendezvousDeviceDelegate : public RendezvousSessionDelegate
```

Comment:
Might be outside the scope of this PR, but I'm wondering if this class should purely be `getPinCode(), open(), close(), and error()` callbacks. The Sending and Receiving of Messages can eventually be handled internally within CHIP as it might be common for most accessories right?

Reply:
I think it's mainly driven by the separation of handling of WiFi credentials on the device side.